src/main/java/com/yelp/nrtsearch/server/luceneserver/search/SearchRequestProcessor.java

```diff
@@ -158,12 +157,12 @@ public static SearchContext buildContextForRequest(
 
     Highlight highlight = searchRequest.getHighlight();
     if (!highlight.getFieldsList().isEmpty()) {
-      verifyHighlights(indexState, highlight);
       HighlightFetchTask highlightFetchTask =
-          new HighlightFetchTask(indexState, searcherAndTaxonomy, query, highlight);
+          new HighlightFetchTask(
+              indexState, searcherAndTaxonomy, query, HighlighterService.getInstance(), highlight);
       contextBuilder.setHighlightFetchTask(highlightFetchTask);
     }
-
+    contextBuilder.setExtraContext(new HashMap<>());
```

Comment:
Is this used already? Do we need to worry about threadsafe for hashmap here now?

Reply:
It is not used yet. We want to use it in plan highlighter to cache highlighter per field.
It shall be thread safe though